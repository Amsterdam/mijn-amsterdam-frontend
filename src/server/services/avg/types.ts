import { ZaakAanvraagDetail } from '../../../universal/types/App.types';
import { SmileSourceResponse, SmileFieldValue } from '../smile/smile-types';

export type AVGResponse = {
  aantal: number;
  verzoeken: AVGRequestFrontend[];
};

export interface AVGRequestFrontend extends ZaakAanvraagDetail {
  identifier: string;
  registratieDatum: string;
  type: string;
  toelichting: string;
  resultaat: string;
  ontvangstDatum: string;
  ontvangstDatumFormatted: string;
  opschortenGestartOp: string;
  datumInBehandeling: string;
  datumAfhandeling: string;
  themas: string;
}

export type SmileAvgResponse = SmileSourceResponse<SmileAVGRequest>;

export type SmileAVGRequest = {
  avgverzoek_id: SmileFieldValue;
  avgverzoek_statusavgverzoek_alias: SmileFieldValue;
  avgverzoek_datumbinnenkomst: SmileFieldValue;
  avgverzoek_typeverzoek: SmileFieldValue;
  avgverzoek_typeafhandelingvaststellen_resultaat: SmileFieldValue | null;
  avgverzoek_omschrijvingvanonderwerp: SmileFieldValue;
  avgverzoek_opschortengestartop: SmileFieldValue | null;
  avgverzoek_datuminbehandeling: SmileFieldValue | null;
  avgverzoek_werkelijkeeinddatum: SmileFieldValue | null;
};

export type SmileAvgThemesResponse = SmileSourceResponse<SmileAvgThemes>;

export type SmileAvgThemes = {
  themaperavgverzoek_avgthema_omschrijving: SmileFieldValue;
  themaperavgverzoek_avgverzoek_id: SmileFieldValue;
};

export type AvgTheme = {
  avgVerzoekId: string | null;
  themaOmschrijving: string | null;
};

export type AvgThemesResponse = {
  verzoeken: AvgTheme[];
};
