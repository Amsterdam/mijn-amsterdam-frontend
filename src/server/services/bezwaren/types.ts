import { GenericDocument, ZaakDetail } from '../../../universal/types';

export type kenmerkKey =
  | 'statustekst'
  | 'statusdatum'
  | 'resultaattekst'
  | 'besluitnr'
  | 'besluitdatum'
  | 'datumintrekking';

export type Kenmerk = { kenmerk: kenmerkKey; bron: string };

export interface BezwaarSourceData {
  uuid: string;
  // Ontvangstdatum
  registratiedatum: string;

  // Bezwaarnummer
  identificatie: string;

  // Onderwerp
  omschrijving: string | null;

  // Specificatie
  toelichting: string | null;

  // Status tekst
  status: string | null;

  // Datum van het resultaat van het bezwaar
  publicatiedatum: string | null;

  // In behandeling
  startdatum: string;

  einddatum: string | null;

  kenmerken: Kenmerk[];
}

export type Bezwaar = {
  identificatie: string;
  uuid: string;
  fetchUrl: string | null;
  startdatum: string;
  ontvangstdatum: string;
  ontvangstdatumFormatted: string | null;
  omschrijving: string | null;
  toelichting: string | null;
  statusdatum: string | null;
  datumResultaat: string | null;
  datumResultaatFormatted: string | null;
  einddatum: string | null;
  primairbesluit: string | null;
  primairbesluitdatum: string | null;
  primairbesluitdatumFormatted: string | null;
  resultaat: string | null;
  steps: never[];
  documenten: BezwaarDocument[];
} & ZaakDetail;

export type BezwaarSourceStatus = {
  url: string;
  uuid: string;
  zaak: string;
  statustype: string;
  datumStatusGezet: string;
  statustoelichting: string;
};

export interface BezwarenSourceResponse<T> {
  count: number;
  next: string;
  previous: string;
  results: T[];
}

export interface BezwaarSourceDocument {
  url: string;
  identificatie: string;
  bronorganisatie: string | null;
  creatiedatum: string;
  titel: string;
  vertrouwelijkheidaanduiding: string;
  auteur: string;
  status: string;
  formaat: string;
  taal: string;
  versie: number;
  beginRegistratie: string;
  bestandsnaam: string;
  inhoud: string;
  bestandsomvang: string | null;
  link: string | null;
  beschrijving: string;
  ontvangstdatum: string;
  verzenddatum: string;
  indicatieGebruiksrecht: string | null;
  verschijningsvorm: string | null;
  ondertekening: string | null;
  integriteit: string | null;
  informatieobjecttype: string;
  locked: boolean;
  dossiertype:
    | 'Online Correspondentie'
    | 'Online Procesdossier'
    | 'Online Besluitvorming'
    | 'Online Aangeleverd';
}

export type BezwaarDocument = GenericDocument & {
  dossiertype: BezwaarSourceDocument['dossiertype'];
};

export interface OctopusApiResponse<T> {
  items: T[];
  count: number;
}
