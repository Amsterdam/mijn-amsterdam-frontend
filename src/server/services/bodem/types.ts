import {
  GenericDocument,
  LinkProps,
  ZaakDetail,
} from '../../../universal/types';

export type Lood365Response = {
  '@onformdata.context': string;
  responsedata: string; // String with a encoded JSON message containing LoodMetingRequestSource object.
};

export type LoodMetingRequestsSource = {
  Requestor: {
    Initials: string;
    Firstname: string;
    Lastname: string;
    Mainphone: string | null;
    Mobilephone: string | null;
    Emailaddress: string;
  };
  Requests: LoodMetingRequestSource[];
};

type LoodMetingRequestSource = {
  Reference: string;
  RequestedOn: string;
  Researchlocations: LoodMetingResearchLocationSource[];
};

type SourceStatus = 'Ontvangen' | 'In behandeling' | 'Afgewezen';

type LoodMetingResearchLocationSource = {
  Reference: string;
  Street: string;
  Housenumber: string;
  Houseletter: string | null;
  Postalcode: string;
  City: string;
  Workorderid: string | null;
  Workordercreatedon: string | null;
  Friendlystatus: SourceStatus;
  Rejectionreason: string | null;
  ReviewedOn: string | null;
  Reportsenton: string | null;
  Reportavailable: boolean;
};

export type LoodMetingen = {
  metingen: LoodMetingFrontend[];
};

export type LoodMetingStatus = SourceStatus | 'Afgehandeld';

export interface LoodMetingFrontend extends ZaakDetail {
  adres: string;
  datumAanvraag: string; // RequestedOn
  datumAanvraagFormatted: string;
  datumInbehandeling: string | null; // Workordercreatedon
  datumAfgehandeld: string | null; // Reportsenton
  datumBeoordeling: string | null; // ReviewedOn
  status: LoodMetingStatus;
  kenmerk: string;
  aanvraagNummer: string;
  rapportBeschikbaar: boolean;
  link: LinkProps;
  redenAfwijzing: string | null;
  rapportId: string | null;
  document: GenericDocument | null;
}

export type LoodMetingDocument = {
  filename: string;
  mimetype: string;
  documentbody: string;
};
