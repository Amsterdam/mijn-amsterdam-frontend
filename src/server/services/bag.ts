import { LatLngLiteral } from 'leaflet';

import { apiErrorResult } from '../../universal/helpers/api';
import {
  getBagResult,
  getBagSearchAddress,
  getLatLonByAddress,
} from '../../universal/helpers/bag';
import { Adres } from '../../universal/types';
import { getApiConfig } from '../helpers/source-api-helpers';
import { requestData } from '../helpers/source-api-request';

export interface BAGData {
  latlng: LatLngLiteral | null;
  address?: Adres | null;
  bagNummeraanduidingId?: string | null;
  profileType: ProfileType;
}

export async function fetchBAG(
  requestID: RequestID,
  sourceAddress: Adres | null
) {
  if (!sourceAddress) {
    return apiErrorResult('Could not query BAG, no address supplied.', null);
  }

  const searchAddress = getBagSearchAddress(sourceAddress);

  if (!searchAddress) {
    return apiErrorResult(`Kon geen correct zoek adres opmaken.`, null);
  }

  const params = { q: searchAddress, features: 2 }; // features=2 is een Feature flag zodat ook locaties in Weesp worden weergegeven.
  const config = getApiConfig('BAG', {
    params,
    cacheKey: `${requestID}-${searchAddress}`,
    transformResponse: (responseData) => {
      const isWeesp = sourceAddress.woonplaatsNaam === 'Weesp';

      const latlng = getLatLonByAddress(
        responseData?.results,
        searchAddress,
        isWeesp
      );

      const bagResult = getBagResult(
        responseData?.results,
        searchAddress,
        isWeesp
      );

      return {
        latlng,
        address: sourceAddress,
        bagNummeraanduidingId: bagResult?.landelijk_id ?? null,
      };
    },
  });
  return requestData<BAGData>(config, requestID);
}
