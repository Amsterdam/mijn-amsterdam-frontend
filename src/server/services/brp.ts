import { differenceInCalendarDays, differenceInMonths } from 'date-fns';
import { generatePath } from 'react-router-dom';

import { AppRoutes } from '../../universal/config/routes';
import { Themas } from '../../universal/config/thema';
import {
  ApiResponse,
  ApiSuccessResponse,
  apiDependencyError,
  apiSuccessResult,
} from '../../universal/helpers/api';
import { defaultDateFormat } from '../../universal/helpers/date';
import {
  BRPData,
  BRPDataFromSource,
  MyNotification,
} from '../../universal/types';
import { AuthProfileAndToken } from '../auth/auth-types';
import { getApiConfig } from '../helpers/source-api-helpers';
import { requestData } from '../helpers/source-api-request';
import { BffEndpoints } from '../routing/bff-routes';
import { generateFullApiUrlBFF } from '../routing/route-helpers';

const DAYS_BEFORE_EXPIRATION = 120;
const MONTHS_TO_KEEP_NOTIFICATIONS = 12;

// You need a id valid for 2 more month when returning to NL. So everyone with a expiring id before this date will probably want to renew their id before the summer holidays --> this creates a peak at the Stadsloket from march onwards.
const ID_BEWIJS_PEAK_DATE_START = new Date('2024-03-01');
const ID_BEWIJS_PEAK_DATE_END = new Date('2024-10-01');

const BrpDocumentTitles: Record<string, string> = {
  paspoort: 'paspoort',
  'europese identiteitskaart': 'ID-kaart',
  'nederlandse identiteitskaart': 'ID-kaart',
  rijbewijs: 'rijbewijs',
};

const BrpDocumentCallToAction: Record<string, string> = {
  paspoort: 'https://www.amsterdam.nl/burgerzaken/paspoort-id-kaart-aanvragen/',
  'europese identiteitskaart':
    'https://www.amsterdam.nl/burgerzaken/paspoort-id-kaart-aanvragen/',
  'nederlandse identiteitskaart':
    'https://www.amsterdam.nl/burgerzaken/paspoort-id-kaart-aanvragen/',
  rijbewijs: 'https://www.amsterdam.nl/burgerzaken/rijbewijs/',
};

export function transformBRPNotifications(data: BRPData, compareDate: Date) {
  const adresInOnderzoek = data?.persoon?.adresInOnderzoek;
  const isOnbekendWaarheen = data?.persoon?.vertrokkenOnbekendWaarheen || false;
  const dateLeft = data?.persoon?.datumVertrekUitNederland
    ? defaultDateFormat(data?.persoon.datumVertrekUitNederland)
    : 'Onbekend';

  const notifications: MyNotification[] = [];

  // Expired and not expired more than 12 months
  const expiredDocuments = data.identiteitsbewijzen?.filter((document) => {
    const dateExpired = new Date(document.datumAfloop);
    return (
      dateExpired < compareDate &&
      differenceInMonths(compareDate, dateExpired) <=
        MONTHS_TO_KEEP_NOTIFICATIONS
    );
  });

  const willExpireSoonDocuments = data.identiteitsbewijzen?.filter(
    (document) => {
      const days = differenceInCalendarDays(
        new Date(document.datumAfloop),
        compareDate
      );

      return days <= DAYS_BEFORE_EXPIRATION && days > 0;
    }
  );

  const documentsExpiringDuringPeak = data.identiteitsbewijzen?.filter(
    (document) => {
      const afloop = new Date(document.datumAfloop);
      return (
        afloop >= ID_BEWIJS_PEAK_DATE_START && afloop <= ID_BEWIJS_PEAK_DATE_END
      );
    }
  );

  if (expiredDocuments?.length) {
    expiredDocuments.forEach((document) => {
      const docTitle =
        BrpDocumentTitles[document.documentType] || document.documentType;
      notifications.push({
        thema: Themas.BURGERZAKEN,
        datePublished: compareDate.toISOString(),
        hideDatePublished: true,
        isAlert: true,
        id: `${docTitle}-datum-afloop-verstreken`,
        title: `Uw ${docTitle} is verlopen`,
        description: `Sinds ${defaultDateFormat(
          document.datumAfloop
        )} is uw ${docTitle} niet meer geldig.`,
        link: {
          to: BrpDocumentCallToAction[document.documentType],
          title: `Vraag uw nieuwe ${docTitle} aan`,
        },
      });
    });
  }

  if (documentsExpiringDuringPeak?.length) {
    documentsExpiringDuringPeak.forEach((document) => {
      const docTitle =
        BrpDocumentTitles[document.documentType] || document.documentType;
      notifications.push({
        thema: Themas.BURGERZAKEN,
        datePublished: compareDate.toISOString(),
        isAlert: true,
        id: `${document.documentType}-datum-afloop-binnekort`,
        title: `Voorkom vertraging en verleng uw ${docTitle} op tijd`,
        description: `Vanaf maart tot de zomervakantie wordt het erg druk op het Stadsloket. Uw huidige ${docTitle} verloopt op ${defaultDateFormat(
          document.datumAfloop
        )}, vraag uw nieuwe ${docTitle} daarom ruim van te voren aan. Tip: in de ochtend is het rustiger bij het Stadsloket.`,
        link: {
          to: BrpDocumentCallToAction[document.documentType],
          title: `Vraag uw nieuwe ${docTitle} aan`,
        },
      });
    });
  }

  if (!documentsExpiringDuringPeak?.length && willExpireSoonDocuments?.length) {
    willExpireSoonDocuments.forEach((document) => {
      const docTitle =
        BrpDocumentTitles[document.documentType] || document.documentType;
      notifications.push({
        thema: Themas.BURGERZAKEN,
        datePublished: compareDate.toISOString(),
        isAlert: true,
        hideDatePublished: true,
        id: `${document.documentType}-datum-afloop-binnekort`,
        title: `Uw ${docTitle} verloopt binnenkort`,
        description: `Vanaf ${defaultDateFormat(
          document.datumAfloop
        )} is uw ${docTitle} niet meer geldig.`,
        link: {
          to: BrpDocumentCallToAction[document.documentType],
          title: `Vraag uw nieuwe ${docTitle} aan`,
        },
      });
    });
  }

  if (adresInOnderzoek) {
    notifications.push({
      thema: Themas.BRP,
      datePublished: compareDate.toISOString(),
      isAlert: true,
      id: 'brpAdresInOnderzoek',
      title: 'Adres in onderzoek',
      description:
        adresInOnderzoek === '080000'
          ? 'Op dit moment onderzoeken wij of u nog steeds woont op het adres waar u ingeschreven staat.'
          : 'Op dit moment onderzoeken wij op welk adres u nu woont.',
      link: {
        to: AppRoutes.BRP,
        title: 'Meer informatie',
      },
    });
  }

  if (isOnbekendWaarheen) {
    notifications.push({
      thema: Themas.BRP,
      datePublished: compareDate.toISOString(),
      isAlert: true,
      id: 'brpVertrokkenOnbekendWaarheen',
      title: 'Vertrokken Onbekend Waarheen (VOW)',
      description: `U staat sinds ${dateLeft} in de Basisregistratie Personen (BRP) geregistreerd als 'vertrokken onbekend waarheen'.`,
      link: {
        to: AppRoutes.BRP,
        title: 'Meer informatie',
      },
    });
  }

  return notifications;
}

export function transformBRPData(
  responseData: ApiSuccessResponse<BRPDataFromSource>
) {
  const responseContent = responseData.content;
  if (responseContent) {
    responseContent.fetchUrlAantalBewoners = generateFullApiUrlBFF(
      BffEndpoints.MKS_AANTAL_BEWONERS,
      {
        addressKeyEncrypted:
          responseContent?.adres?._adresSleutel ?? '--no-address-key--',
      }
    );
  }
  if (Array.isArray(responseContent?.identiteitsbewijzen)) {
    // Transform Identiteitsbewijzen
    Object.assign(responseContent, {
      identiteitsbewijzen: responseContent.identiteitsbewijzen.map(
        (document) => {
          const route = generatePath(AppRoutes['BURGERZAKEN/ID-KAART'], {
            id: document.id,
          });
          return Object.assign({}, document, {
            title:
              BrpDocumentTitles[document.documentType] || document.documentType,
            datumAfloop: document.datumAfloop,
            datumUitgifte: document.datumUitgifte,
            link: {
              to: route,
              title: document.documentType,
            },
          });
        }
      ),
    });
  }

  return responseContent as BRPData;
}

export async function fetchBRP(
  requestID: RequestID,
  authProfileAndToken: AuthProfileAndToken
) {
  const config = getApiConfig('BRP', {
    transformResponse: transformBRPData,
  });

  return requestData<BRPData>(config, requestID, authProfileAndToken);
}

export async function fetchBrpNotifications(
  requestID: RequestID,
  authProfileAndToken: AuthProfileAndToken
) {
  const BRP = await fetchBRP(requestID, authProfileAndToken);

  if (BRP.status === 'OK') {
    return apiSuccessResult({
      notifications: transformBRPNotifications(BRP.content, new Date()),
    });
  }
  return apiDependencyError({ BRP });
}

export async function fetchAantalBewoners(
  requestID: RequestID,
  authProfileAndToken: AuthProfileAndToken,
  addressKeyEncrypted: string
) {
  const url = `${process.env.BFF_MKS_API_BASE_URL}/brp/aantal_bewoners`;

  return requestData(
    {
      url,
      method: 'POST',
      passthroughOIDCToken: true,
      data: {
        addressKey: addressKeyEncrypted,
      },
      transformResponse: (responseData: ApiResponse<string>) => {
        if (responseData.status === 'OK') {
          return responseData.content;
        }
        return responseData.content;
      },
    },
    requestID,
    authProfileAndToken
  );
}
