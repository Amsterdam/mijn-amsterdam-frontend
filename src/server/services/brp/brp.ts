import { ONE_HOUR_MS } from '../../config/app';
import { getFromEnv } from '../../helpers/env';
import { fetchAuthTokenHeader } from '../ms-oauth/oauth-token';

const TOKEN_VALIDITY_PERIOD = 24 * ONE_HOUR_MS;
const PERCENTAGE_DISTANCE_FROM_EXPIRY = 0.1;

function fetchBenkBrpTokenHeader() {
  return fetchAuthTokenHeader(
    {
      serviceID: 'BRP',
      tokenValidityMS:
        TOKEN_VALIDITY_PERIOD * (1 - PERCENTAGE_DISTANCE_FROM_EXPIRY),
    },
    {
      clientID: getFromEnv('BFF_BENK_BRP_CLIENT_ID') ?? '',
      clientSecret: getFromEnv('BFF_BENK_BRP_CLIENT_SECRET') ?? '',
      tenantID: getFromEnv('BFF_BENK_BRP_TENANT') ?? '',
      scope: `${getFromEnv('BFF_BENK_BRP_APPLICATION_ID')}/.default`,
    }
  );
}
