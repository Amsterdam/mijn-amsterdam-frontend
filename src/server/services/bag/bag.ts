import { BAGQueryParams, type BAGData, type BAGSourceData } from './bag.types';
import {
  apiErrorResult,
  type ApiResponse,
} from '../../../universal/helpers/api';
import { getLatLngCoordinates } from '../../../universal/helpers/bag';
import { getApiConfig } from '../../helpers/source-api-helpers';
import { requestData } from '../../helpers/source-api-request';
import type { Adres } from '../profile/brp.types';

function transformBagResponse(
  sourceAddress: Adres,
  responseData: BAGSourceData
): BAGData | null {
  const data = responseData._embedded?.adresseerbareobjecten;
  if (!data?.length) {
    return null;
  }

  // Multiple items can be found, but only the first we take as relevant.
  const firstItem = data[0];

  const latlng = getLatLngCoordinates(
    firstItem.adresseerbaarObjectPuntGeometrieWgs84.coordinates
  );

  return {
    latlng,
    address: sourceAddress,
    bagAddress: firstItem,
    bagNummeraanduidingId: firstItem.identificatie,
  };
}

export async function fetchBAG(
  sourceAddress: Adres | null
): Promise<ApiResponse<BAGData>> {
  if (!sourceAddress?.straatnaam || !sourceAddress.huisnummer) {
    return apiErrorResult('Could not query BAG, no address supplied.', null);
  }

  const params: BAGQueryParams = {
    openbareruimteNaam: sourceAddress.straatnaam,
    huisnummer: sourceAddress.huisnummer,
    huisletter: sourceAddress.huisletter || undefined,
  };

  const config = getApiConfig('BAG', {
    params,
    transformResponse(responseData: BAGSourceData): BAGData | null {
      return transformBagResponse(sourceAddress, responseData);
    },
  });
  return requestData<BAGData>(config);
}
