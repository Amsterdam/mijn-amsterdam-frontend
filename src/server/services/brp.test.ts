import { transformBRPData, transformBRPNotifications } from './brp';
import brpData from '../../../mocks/fixtures/brp.json';
import { ApiSuccessResponse } from '../../universal/helpers/api';
import { getBagSearchAddress } from '../../universal/helpers/bag';
import { getFullAddress } from '../../universal/helpers/brp';
import { BRPDataFromSource } from '../../universal/types/brp';

const brpDataTyped = brpData as ApiSuccessResponse<BRPDataFromSource>;
const {
  content: { adres },
} = brpDataTyped;

describe('BRP data api + transformation', () => {
  it('should construct a bag search addresss', () => {
    expect(getBagSearchAddress(adres)).toBe('Weesperstraat 113');
  });

  it('should construct a complete addresss', () => {
    expect(
      getFullAddress({ ...adres, huisletter: 'X', huisnummertoevoeging: 'h' })
    ).toBe('Weesperstraat 113 X h');
  });

  it('should transform the source data', () => {
    expect(transformBRPData(brpDataTyped)).toMatchSnapshot();
  });

  it('should transform the source data into notifications', () => {
    const data = transformBRPData(brpDataTyped);
    expect(
      transformBRPNotifications(data, new Date(2020, 3, 23))
    ).toMatchSnapshot();
  });
});
