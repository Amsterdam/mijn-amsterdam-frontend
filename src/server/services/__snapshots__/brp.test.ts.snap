// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BRP data api + transformation > should transform the source data 1`] = `
{
  "adres": {
    "_adresSleutel": "xxxcasdasfada",
    "adresType": "correspondentie",
    "begindatumVerblijf": "1967-01-01",
    "einddatumVerblijf": "1967-01-01",
    "huisletter": null,
    "huisnummer": "113",
    "huisnummertoevoeging": null,
    "landnaam": "Nederland",
    "postcode": "1018 DN",
    "straatnaam": "Weesperstraat",
    "woonplaatsNaam": "Amsterdam",
  },
  "adresHistorisch": [
    {
      "adresType": "woon",
      "begindatumVerblijf": "2005-09-01",
      "einddatumVerblijf": "2012-08-01",
      "huisletter": null,
      "huisnummer": "9",
      "huisnummertoevoeging": "H",
      "landnaam": "Nederland",
      "postcode": "1098 NK",
      "straatnaam": "Ampèrestraat",
      "woonplaatsNaam": "Amsterdam",
    },
    {
      "adresType": "woon",
      "begindatumVerblijf": "1990-01-01",
      "einddatumVerblijf": "2005-09-01",
      "huisletter": null,
      "huisnummer": "9",
      "huisnummertoevoeging": "3",
      "landnaam": "Nederland",
      "postcode": "1098 AA",
      "straatnaam": "Middenweg",
      "woonplaatsNaam": "Amsterdam",
    },
  ],
  "fetchUrlAantalBewoners": "http://bff-api-host/api/v1/service/mks/aantal-bewoners/xxxcasdasfada",
  "identiteitsbewijzen": [
    {
      "datumAfloop": "2025-10-15",
      "datumUitgifte": "2015-10-15",
      "documentNummer": "PP57XKG54",
      "documentType": "paspoort",
      "id": "een-hash-van-documentnummer-1",
      "link": {
        "title": "paspoort",
        "to": "/burgerzaken/id-kaart/een-hash-van-documentnummer-1",
      },
      "title": "paspoort",
    },
    {
      "datumAfloop": "2020-09-11",
      "datumUitgifte": "2010-09-11",
      "documentNummer": "IE9962819",
      "documentType": "europese identiteitskaart",
      "id": "een-hash-van-documentnummer-2",
      "link": {
        "title": "europese identiteitskaart",
        "to": "/burgerzaken/id-kaart/een-hash-van-documentnummer-2",
      },
      "title": "ID-kaart",
    },
  ],
  "kinderen": [
    {
      "bsn": "123456780",
      "geboortedatum": "2016-07-08",
      "geslachtsaanduiding": "M",
      "geslachtsnaam": "Kosterijk",
      "overlijdensdatum": null,
      "voornamen": "Yassine",
      "voorvoegselGeslachtsnaam": null,
    },
    {
      "bsn": "123456780",
      "geboortedatum": "2019-07-08",
      "geslachtsaanduiding": "M",
      "geslachtsnaam": "Kosterijk",
      "overlijdensdatum": null,
      "voornamen": "Marwan",
      "voorvoegselGeslachtsnaam": null,
    },
  ],
  "kvkNummer": "123123123",
  "ouders": [
    {
      "bsn": "123456780",
      "geboortedatum": "1920-01-01",
      "geboortelandnaam": "Nederland",
      "geboorteplaatsnaam": "Lochem",
      "geslachtsnaam": "Beemsterboer",
      "indicatieGeboortedatum": "M",
      "nationaliteiten": [
        {
          "omschrijving": "Nederlandse",
        },
      ],
      "omschrijvingGeslachtsaanduiding": "Man",
      "opgemaakteNaam": "S. Beemsterboer",
      "overlijdensdatum": null,
      "voornamen": "Senior",
      "voorvoegselGeslachtsnaam": null,
    },
    {
      "bsn": "123456780",
      "geboortedatum": "1920-01-01",
      "geboortelandnaam": "Nederland",
      "geboorteplaatsnaam": "Lochem",
      "geslachtsnaam": "Beemsterboerin",
      "nationaliteiten": [
        {
          "omschrijving": "Nederlandse",
        },
      ],
      "omschrijvingGeslachtsaanduiding": "Vrouw",
      "opgemaakteNaam": "Sa. Beemsterboer",
      "voornamen": "Seniora",
      "voorvoegselGeslachtsnaam": null,
    },
  ],
  "persoon": {
    "aanduidingNaamgebruikOmschrijving": "Eigen geslachtsnaam",
    "adresInOnderzoek": "089999",
    "bsn": "129095205",
    "datumVertrekUitNederland": "1967-01-01",
    "geboortedatum": "1992-09-22",
    "geboortelandnaam": "Nederland",
    "geboorteplaatsnaam": "Lochem",
    "gemeentenaamInschrijving": "Amsterdam",
    "geslachtsnaam": "Beemsterboer",
    "indicatieGeboortedatum": "J",
    "indicatieGeheim": true,
    "mokum": true,
    "nationaliteiten": [
      {
        "omschrijving": "Nederlandse",
      },
      {
        "omschrijving": "Turkse",
      },
    ],
    "omschrijvingAdellijkeTitel": "Hertogin",
    "omschrijvingBurgerlijkeStaat": "Gehuwd",
    "omschrijvingGeslachtsaanduiding": "Man",
    "opgemaakteNaam": "W. Beemsterboer",
    "overlijdensdatum": null,
    "vertrokkenOnbekendWaarheen": true,
    "voornamen": "Wesley",
    "voorvoegselGeslachtsnaam": null,
  },
  "verbintenis": {
    "datumOntbinding": null,
    "datumSluiting": "1999-01-01",
    "landnaamSluiting": "Marokko",
    "persoon": {
      "bsn": "123456780",
      "geboortedatum": "2019-07-08",
      "geslachtsnaam": "Bakker",
      "indicatieGeboortedatum": "D",
      "overlijdensdatum": null,
      "voornamen": "Souad",
      "voorvoegselGeslachtsnaam": null,
    },
    "plaatsnaamSluitingOmschrijving": "Asilah",
    "soortVerbintenis": "H",
    "soortVerbintenisOmschrijving": "Huwelijk",
  },
  "verbintenisHistorisch": [
    {
      "datumOntbinding": "1998-07-08",
      "datumSluiting": "1912-01-01",
      "landnaamSluiting": "Marokko",
      "persoon": {
        "bsn": "123456780",
        "geboortedatum": "2019-07-08",
        "geslachtsnaam": "Bakker",
        "indicatieGeboortedatum": "V",
        "overlijdensdatum": null,
        "voornamen": "Souad",
        "voorvoegselGeslachtsnaam": null,
      },
      "plaatsnaamSluitingOmschrijving": "Asilah",
      "soortVerbintenis": "H",
      "soortVerbintenisOmschrijving": "Huwelijk",
    },
    {
      "datumOntbinding": "2019-07-08",
      "datumSluiting": "",
      "landnaamSluiting": "Marokko",
      "persoon": {
        "bsn": "123456780",
        "geboortedatum": "2019-07-08",
        "geslachtsnaam": "Bakker",
        "overlijdensdatum": null,
        "voornamen": "Souad",
        "voorvoegselGeslachtsnaam": null,
      },
      "plaatsnaamSluitingOmschrijving": "Asilah",
      "soortVerbintenis": "H",
      "soortVerbintenisOmschrijving": "Huwelijk",
    },
    {
      "datumOntbinding": "2006-01-01",
      "datumSluiting": "2000-01-01",
      "landnaamSluiting": "Nederland",
      "persoon": {
        "bsn": "113731681",
        "geboortedatum": "1950-01-06",
        "geboortelandnaam": "Nederland",
        "geboorteplaatsnaam": "Amsterdam",
        "geslachtsnaam": "Wolters",
        "omschrijvingAdellijkeTitel": "Barones",
        "omschrijvingGeslachtsaanduiding": "Vrouw",
        "opgemaakteNaam": null,
        "overlijdensdatum": null,
        "voornamen": "Renée",
        "voorvoegselGeslachtsnaam": null,
      },
      "plaatsnaamSluitingOmschrijving": "Amsterdam",
      "redenOntbindingOmschrijving": "Overlijden één van beide partners",
      "soortVerbintenis": "H",
      "soortVerbintenisOmschrijving": "Huwelijk",
    },
  ],
}
`;

exports[`BRP data api + transformation > should transform the source data into notifications 1`] = `
[
  {
    "datePublished": "2020-04-22T22:00:00.000Z",
    "description": "Op dit moment onderzoeken wij op welk adres u nu woont.",
    "id": "brpAdresInOnderzoek",
    "isAlert": true,
    "link": {
      "title": "Meer informatie",
      "to": "/persoonlijke-gegevens",
    },
    "thema": "BRP",
    "title": "Adres in onderzoek",
  },
  {
    "datePublished": "2020-04-22T22:00:00.000Z",
    "description": "U staat sinds 01 januari 1967 in de Basisregistratie Personen (BRP) geregistreerd als 'vertrokken onbekend waarheen'.",
    "id": "brpVertrokkenOnbekendWaarheen",
    "isAlert": true,
    "link": {
      "title": "Meer informatie",
      "to": "/persoonlijke-gegevens",
    },
    "thema": "BRP",
    "title": "Vertrokken Onbekend Waarheen (VOW)",
  },
]
`;
